superset/assets/javascripts/explorev2/components/controls/Filter.jsx

```diff
@@ -55,7 +58,23 @@ export default class Filter extends React.Component {
     if (event && event.value) {
       value = event.value;
     }
-    this.props.changeFilter(control, value);
+    if (control === 'op') {
+      if (arrayFilterOps.indexOf(this.props.filter.op) !== -1
```

Comment:
This whole section if very opaque now. It probably requires to be either simplified to be more readable or be cut into a block and add a descriptive comment at the top.